File: collect_app/src/test/java/org/odk/collect/android/support/LiveDataTester.java
package org.odk.collect.android.support;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.Observer;

public class LiveDataTester {

    private final FakeLifecycleOwner owner = new FakeLifecycleOwner();

    public <T> LiveData<T> activate(LiveData<T> liveData) {
        liveData.observe(owner, (Observer<Object>) any -> { });
        return liveData;
    }

    public void teardown() {
        owner.destroy();
    }
}

File: testshared/src/main/java/org/odk/collect/testshared/android/LiveDataTester.kt
package org.odk.collect.testshared.android

import androidx.lifecycle.LiveData
import androidx.lifecycle.Observer
import org.odk.collect.testshared.FakeLifecycleOwner

class LiveDataTester {

    private val owner = FakeLifecycleOwner()

    fun <T> activate(liveData: LiveData<T>): LiveData<T> {
        liveData.observe(owner, Observer { })
        return liveData
    }

    fun teardown() {
        owner.destroy()
    }
}
